lib/charms/prometheus_k8s/v1/prometheus_remote_write.py

```diff
@@ -484,11 +490,12 @@ def _push_alerts_to_relation_databag(self, relation: Relation) -> None:
             return
 
         alert_rules = AlertRules(query_type="promql", topology=self.topology)
-        alert_rules.add_path(self._alert_rules_path)
+        if not self._disable_alerts:
+            alert_rules.add_path(self._alert_rules_path)
```

Comment:
Is there any code path you can think of where in the constructor we now flipped from one bool to another, but `_push_alerts_to_relation_databag` isn't called?

In other words
- should the lib update reldata on construction?
- do we need a "recreate the world" function in the lib?

Same question applies to the scrape lib.